### ISSUE TITLE
Improve CI coverage for Salvo

### ISSUE THREAD

**Issue by mum4k** [envoyproxy/envoy-perf#135]
We should migrate our CI pipelines to AZP and make sure they execute all steps including format and coverage.

**Comment by mum4k** [comment]
@gyohuangxin thank you for your interest in helping here. I would be happy to advise on the AZP steps. Those will be "the easy part". First we need to have a script that will be able to execute all the phases of our CI.

I would model this after what we did in Nighthawk, where we have this [do_ci.sh](https://github.com/envoyproxy/nighthawk/blob/main/ci/do_ci.sh) script. This script takes an argument indicating the CI [stage it should execute](https://github.com/envoyproxy/nighthawk/blob/46387e5771aa74248f3fd9c0c6abc58a58e2b621/ci/do_ci.sh#L319).

For Salvo we already have a [similar script](https://github.com/envoyproxy/envoy-perf/blob/main/ci/do_ci.sh), but with much less functionality. We will want at least the following stages executable from the script:
- `check_format` to check the format of all files.
- `fix_format` to automatically apply formatting fixes (this is for local execution).
- `build` to bazel build all our targets. This already exists.
- `test` to bazel test (execute unit tests). This already exists.
- `coverage` to calculate our test coverage and compare it to a threshold.
- `integration` to execute the simplest benchmark with Salvo and make sure it passes.

And optionally:
- `sanitizers` to execute with ASAN, TSAN and MSAN. While this isn't important right away, we are likely to introduce concurrency and more complexity once we start actively developing here, so these would be helpful.

Once we have the script that can execute these phases, porting it onto AZP will be the next step.

**Comment by gyohuangxin** [comment]
@mum4k Thank you, it helps me understand what I should do. I will implement it refer to Nighthawk scripts.

**Comment by mum4k** [comment]
Thank you for your help here @gyohuangxin, please feel free to reach out or on slack with any questions or suggestions.

**Comment by mum4k** [comment]
@gyohuangxin after #136 will be in, you could write an AZP pipeline. You could use the one defined by Nighthawk as an example:

https://github.com/envoyproxy/nighthawk/tree/main/.azure-pipelines

We just need a skeleton of the pipeline in with at least one check. Once that will get submitted, I will configure AZP to start executing and we can open a new PR that will define all the checks in AZP.

As for the integration steps - you indicated that you would like more details / discuss more. Can you contact me on slack with the questions?
